tests/unit/find-record-test.ts

```diff
@@ -10,57 +10,99 @@ import { module, test } from 'qunit';
 import { setupRenderingTest, setupTest } from 'ember-qunit';
 
 import { findRecord } from 'ember-data-resources';
-import { IdRequiredError } from 'ember-data-resources/-private/resources/errors';
+import { IdRequiredError, IdTypeError } from 'ember-data-resources/-private/resources/errors';
 
 import { setupMockData } from './-mock-data';
 
+import type { Id } from 'ember-data-resources/-private/resources/types';
+
 module('findRecord', function (hooks) {
   setupMockData(hooks);
 
   module('in js', function (hooks) {
     setupTest(hooks);
 
-    test('it works', async function (assert) {
+    test('it works when callback returns id', async function (assert) {
       class Blog extends Model {
         @attr name: string | undefined;
       }
 
       this.owner.register('model:blog', Blog);
 
       class Test {
-        @tracked id = 1;
+        @tracked id: Id = 1;
```

Comment:
is this type annotation needed?

Reply:
Without this, it would be considered a `number`, and `id = '2'` below will be marked as a type error.

Reply:
ah, fair